### PR TITLE
Image Customizer: Add doc for building custom packages.

### PR DESCRIPTION
**Pull request by cwize1**
Add a quick guide for how to build custom packages for Azure Linux. This guide uses a normal Azure Linux core container and directly uses `rpmbuild`, instead of using the Azure Linux toolkit, since it is a lot simpler.

---

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

###### Does this affect the toolchain?  

NO

###### Test Methodology

n/a

### CODE REVIEW

File: toolkit/tools/imagecustomizer/docs/building-packages.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+2. Create a file called `samplepackage.spec` with the following contents:
+
+   ```rpmspec
+   Summary:        A sample package
+   Name:           samplepackage
+   Version:        0.0.1
+   Release:        1%{?dist}
+   Vendor:         Contoso
+   License:        Proprietary
```
Comment (dmcilvaney) [inline toolkit/tools/imagecustomizer/docs/building-packages.md:25]: might be illustrative to add a `BuildRequires: words` or something.

File: toolkit/tools/imagecustomizer/docs/building-packages.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Building custom packages
+
+This is a guide on how to build your own packages for Azure Linux.
```
Comment (neha170) [inline toolkit/tools/imagecustomizer/docs/building-packages.md:3]: I would reword this to (something around image)
`This is a guide on how to build your own packages to add a custom image in Azure Linux.`

File: toolkit/tools/imagecustomizer/docs/building-packages.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+   Version:        0.0.1
+   Release:        1%{?dist}
+   Vendor:         Contoso
+   License:        Proprietary
+
+   Source0:        samplescript.sh
+
+   %description
+   A very descriptive description for the sample package.
+
```
Comment (neha170) [inline toolkit/tools/imagecustomizer/docs/building-packages.md:31]: Missing prep and build section (can add blank ones for completeness)

File: toolkit/tools/imagecustomizer/docs/building-packages.md (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizer/docs/building-packages.md]: `BuildRequires` is an odd one, because it doesn't really do anything useful in this situation. Any build dependencies still have to be added to the `Dockerfile` regardless of if they are specified in the `.spec` file.

File: toolkit/tools/imagecustomizer/docs/building-packages.md (hunk 3 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizer/docs/building-packages.md:35]: Fixed.

File: toolkit/tools/imagecustomizer/docs/building-packages.md (hunk 2 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizer/docs/building-packages.md]: Fixed.

File: toolkit/tools/imagecustomizer/docs/building-packages.md (hunk 1 above)
Comment (cwize1) [reply toolkit/tools/imagecustomizer/docs/building-packages.md]: Though, I did add a `Requires:` since those are likely useful to users.

**Review by bfjelds — APPROVED** [review]

**Review by gmileka — APPROVED** [review]
